tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:engine_build_configs/engine_build_configs.dart';
+
+import 'environment.dart';
+
+/// A function that returns true or false when given a [BuildConfig] and its
+/// name.
+typedef ConfigFilter = bool Function(String name, BuildConfig config);
+
+/// A function that returns true or false when given a [BuildConfig] name
+/// and a [GlobalBuild].
+typedef BuildFilter = bool Function(String configName, GlobalBuild build);
+
+/// Returns a filtered copy of [input] filtering out configs where test
+/// returns false.
+Map<String, BuildConfig> filterBuildConfigs(
+    Map<String, BuildConfig> input, ConfigFilter test) {
+  final Map<String, BuildConfig> r = <String, BuildConfig>{};
+  input.forEach((String name, BuildConfig config) {
+    if (!test(name, config)) {
+      return;
+    }
+    r[name] = config;
+  });
+  return r;
+}
+
+/// Returns a copy of [input] filtering out configs that are not runnable
+/// on the current platform.
+Map<String, BuildConfig> runnableBuildConfigs(
+    Environment env, Map<String, BuildConfig> input) {
+  return filterBuildConfigs(input, (String name, BuildConfig config) {
+    return config.canRunOn(env.platform);
+  });
+}
+
+List<GlobalBuild> filterBuilds(
+    Map<String, BuildConfig> input, BuildFilter test) {
+  final List<GlobalBuild> r = <GlobalBuild>[];
```

Comment:
Ditto here, I'd just use the inline-for:
https://dart.dev/language/collections#control-flow-operators

Reply:
Done.